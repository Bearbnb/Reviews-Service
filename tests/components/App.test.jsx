import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import App from '../../client/components/app';

describe('App Component', () => {
  it('App should render App to the DOM', () => {
	  const wrapper = shallow(<App />);
	  expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call componentDidMount once', () => {
    const spy = jest.spyOn(App.prototype, 'componentDidMount');
    const wrapper = mount(<App />);

    wrapper.instance().componentDidMount();
    expect(spy).toHaveBeenCalled();
  });
});
