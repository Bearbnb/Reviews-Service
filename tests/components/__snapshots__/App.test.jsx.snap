// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App should render App to the DOM 1`] = `
<div
  className="App"
>
  <ReviewSummary
    onClick={[Function]}
    resetHomes={[Function]}
    reviews={Array []}
    searched={false}
  />
  <ReviewList
    host={null}
    reviews={Array []}
  />
</div>
`;
